### PR TITLE
Add generic update and clone extension methods

### PR DESCRIPTION
**Pull request by sigurdm**
Still WIP.
Cannot be released before we have dart 2.6.

### CODE REVIEW

**Comment by listepo** [comment]
@sigurdm we have dart 2.6 )

**Comment by listepo** [comment]
@sigurdm any news?

**Comment by zaher-recode** [comment]
Sorry I know you guys are busy, 
but any news, it seems to be more important now with the issue generated by CopyWith method in latest flutter 1.20 version

File: protobuf/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,20 @@
...
+
+  One major gotcha is that if you import your proto files with a prefix these
+  methods will not appear on your instances.
+  Alternatively you can add
+
+  ```
+  import 'package:protobuf/protobuf.dart' show GeneratedMessageGenericExtensions;
+  ```
+
+  To get the extension methods in scope.
```
Comment (osa1) [inline protobuf/CHANGELOG.md:16]: In `protobuf/README.md` we say users typically do not need to import `protobuf`, so we should update it mentioning this extension method.

**Comment by osa1** [comment]
@zaher-recode 

> it seems to be more important now with the issue generated by CopyWith method in latest flutter 1.20 version

Do you have the link to this issue?

**Comment by osa1** [comment]
Closing this as apparently we implemented the same stuff in #394 (merged in 2020), but with different method names. We also added the `@Deprecated` annotation.

`update` in this PR was merged as `rebuild`.
`copy` in this PR was merged as `deepCopy`.

**Pull request closed without merging**